libp2p/muxers/mplex/mplex.nim

```diff
@@ -92,12 +92,15 @@ method handle*(m: Mplex) {.async, gcsafe.} =
             let stream = newConnection(channel)
             stream.peerInfo = m.connection.peerInfo
 
-            # cleanup channel once handler is finished
-            # stream.closeEvent.wait().addCallback(
-            #   proc(udata: pointer) =
-            #       asyncCheck cleanupChann(m, channel, initiator))
-
-            asyncCheck m.streamHandler(stream)
+            proc handler() {.async.} =
+              tryAndWarn "mplex channel handler":
+                await m.streamHandler(stream)
+              # TODO closing stream
```

Comment:
This is due to the fact that channels handle their own lifecycle, closing is initiated with the `close` or `reset` messages, but the channel will not close until it has read/written everything, because it emulates half-closed behavior in TCP.

Reply:
OK so basically it's working as intended?
If so I will add a better comment